File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.cpp
class Solution {
public:
    int numSteps(string s) {
        int ans = 0;
        while(s.length()>1){
            if(s.back()=='1'){
                int j = s.size()-1;
                while(j>=0 && s[j]=='1'){
                    s[j] = '0';
                    j-=1;
                }
                if(j<0){
                    s.insert(s.begin(),'1');
                }
                else{
                    s.insert(s.begin()+j, '1'); 
                    s.pop_back();
                }
            }
            else{
                s.pop_back();
            }
            ans+=1;
        }
        return ans;
    }
};

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        ans = 0
        while len(s)>1:
            if s[-1] == '1':
                j = len(s)-1
                while(j>=0 and s[j]=='1'):
                    s = s[:j] + '0' + s[j+1:]
                    j-=1
                if j<0:
                    s = '1' + s[:]
                else:
                    s = s[0:j] + '1' + s[j+1:]
            else:
                s = s[:-1]
            ans +=1
        return ans


